packages/frontend/core/src/modules/peek-view/view/modal-container.tsx

```diff
@@ -214,10 +212,9 @@ export const PeekViewModalContainer = forwardRef<
           opacity: [0, 1],
           duration: 100,
         },
-      })
-        .then(() => animateControls(true))
-        .catch(console.error);
+      }).catch(console.error);
     }, 0);
+    setTimeout(() => animateControls(true), 400 - 230 + 150);
```

Comment:
any specific meaning of this timeout number?

Reply:
To make sure  the end time of the `controls` animation is `150ms` later than the `modal`, regardless of the actual `transition-duration` of the controls.

Reply:
Can we listen to animationend event instead?

Reply:
Yes it was triggered after dialog animation finished before, but we just need a delay here to start animating controls.

![CleanShot 2024-07-23 at 12.28.40@2x.png](https://graphite-user-uploaded-assets-prod.s3.amazonaws.com/LakojjjzZNf6ogjOVwKE/c0c4df2e-de7d-447a-b070-92aa6e0e57bd.png)

Reply:
add a comment incase we forget this magic number

Reply:
Added

```ts
  /**
   * ### Animation timeline:
   * ```plain
   *                                      150ms
   *                                   ⎮ - - - - ⎮
   * dialog:     +--------400ms--------+
   * controls:               +-------230ms-------+
   *             ⎮ - - - - - ⎮
   *            controls delay =
   *             400 - 230 + 150
   * ```
   */
```